build.zig

```diff
@@ -0,0 +1,459 @@
+    const enable_dynamic_loading = b.option(
+        bool, "enable_dynamic_loading",
+        "Enable tracing of dynamic library data roots") orelse true;
+    const enable_register_main_static_data = b.option(
+        bool, "enable_register_main_static_data",
+        "Perform the initial guess of data root sets") orelse true;
+    const enable_checksums = b.option(
+        bool, "enable_checksums",
+        "Report erroneously cleared dirty bits") orelse false;
+    const enable_werror = b.option(
+        bool, "enable_werror",
+        "Pass -Werror to the C compiler (treat warnings as errors)")
+        orelse false;
+//    const enable_single_obj_compilation = b.option(
+//        bool, "enable_single_obj_compilation",
+//        "Compile all libgc source files into single .o") orelse false;
+//    const disable_single_obj_compilation = b.option(
+//        bool, "disable_single_obj_compilation",
+//        "Compile each libgc source file independently") orelse false;
+    const enable_handle_fork = b.option(
+        bool, "enable_handle_fork",
+        "Attempt to ensure a usable collector after fork()") orelse true;
+    const disable_handle_fork = b.option(
+        bool, "disable_handle_fork",
+        "Prohibit installation of pthread_atfork() handlers") orelse false;
+//    const enable_emscripten_asyncify = b.option(
+//        bool, "enable_emscripten_asyncify",
+//        "Use Emscripten asyncify feature") orelse false;
+    const install_headers = b.option(
+        bool, "install_headers",
+        "Install header and pkg-config metadata files") orelse true;
+//    const with_libatomic_ops = b.option(
+//        bool, "with_libatomic_ops",
+//        "Use an external libatomic_ops") orelse false;
+//    const without_libatomic_ops = b.option(
+//        bool, "without_libatomic_ops",
+//        "Use atomic_ops.h in libatomic_ops/src") orelse false;
+
+    var lib = b.addStaticLibrary(.{
+        .name = "gc",
+        .target = target,
+        .optimize = optimize,
+    });
+    if (build_shared_libs) {
+        lib = b.addSharedLibrary(.{
+            .name = "gc",
+            .target = target,
+            .optimize = optimize,
+        });
+    }
+
+    var source_files = std.ArrayList([]const u8).init(b.allocator);
+    defer source_files.deinit();
+    var flags = std.ArrayList([]const u8).init(b.allocator);
+    defer flags.deinit();
+
+    source_files.appendSlice(&.{
+        "allchblk.c",
+        "alloc.c",
+        "blacklst.c",
+        "dbg_mlc.c",
+        "dyn_load.c",
+        "finalize.c",
+        "headers.c",
+        "mach_dep.c",
+        "malloc.c",
+        "mallocx.c",
+        "mark.c",
+        "mark_rts.c",
+        "misc.c",
+        "new_hblk.c",
+        "obj_map.c",
+        "os_dep.c",
+        "ptr_chck.c",
+        "reclaim.c",
+        "typd_mlc.c",
+    }) catch unreachable;
+
+    var have_getcontext = true;
+    if (t.abi.isMusl()) {
```

Comment:
Unavailability of feature checking is, of course, a bad thing of Zig or at least of our initial version of the script.

Reply:
Agreed. Not sure what the idiomatic way is. I want to look into that more but I don't think it should block this PR... We can always improve later on, right?

Reply:
Right. I hope today I will perform testing and final review.